### PR TITLE
Create libraries to handle the server and clients

### PR DESCRIPTION
**Pull request by rooneyshuman**

### CODE REVIEW

File: mancala/client/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+use std::io::{Read, Write};
+use std::net::TcpStream;
+use std::str::from_utf8;
+
+pub fn run_client() {
+    match TcpStream::connect("localhost:3333") {
```
Comment (logyball) [inline mancala/client/src/lib.rs]: can we make the host/port a variable?

File: mancala/server/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                stream.peer_addr().unwrap()
+            );
+            stream.shutdown(Shutdown::Both).unwrap();
+            false
+        }
+    } {}
+}
+
+pub fn run_server() {
+    let listener = TcpListener::bind("0.0.0.0:3333").unwrap();
```
Comment (logyball) [inline mancala/server/src/lib.rs]: here as well

**Review by logyball — CHANGES_REQUESTED** [review]
I think we could make the directory structure look like this:

```
- Client/
    - src/
        - lib/
            - library files....
        - main.rs
    - cargo.toml
    - cargo.toml.lock
- Server/
    - src/
        - lib/
            - library files....
        - main.rs
    - cargo.toml
    - cargo.toml.lock
- Protocol/
    - src/
        -library files...
    - cargo.toml
    - cargo.toml.lock
- .git
- readme/etc
```

That way, we could import things something like this: 

In client, import the necessary client modules into `client/src/main.rs` and something similar for server, then import the game protocol as an external crate into both server and clint

**Comment by rooneyshuman** [comment]
According to rust docs, cargo needs a particular package layout. (https://doc.rust-lang.org/cargo/guide/project-layout.html).
I've updated the layout to the following:
```
- client/
    - examples/
        - sample_client.rs
    - src/
        - lib.rs
        - main.rs
    - Cargo.toml
    - Cargo.lock
- server/
    - examples/
        - sample_server.rs
    - src/
        - lib.rs
        - main.rs
    - Cargo.toml
    - Cargo.lock
- protocol/
    - src/
        - main.rs
    - Cargo.toml
    - Cargo.lock
- .git
- readme/etc
```

File: mancala/client/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "client"
+version = "0.1.0"
+authors = ["rooneyshuman <rooneyshuman@gmail.com>"]
```
Comment (logyball) [inline mancala/client/Cargo.toml:4]: gonna have to add myself here

File: mancala/server/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "server"
+version = "0.1.0"
+authors = ["rooneyshuman <rooneyshuman@gmail.com>"]
```
Comment (logyball) [inline mancala/server/Cargo.toml:4]: and here

**Review by logyball — APPROVED** [review]

**Pull request merged**